pkg/alerter/alerter_test.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alerter
+
+import (
+	"context"
+	"testing"
+
+	configpb "github.com/GoogleCloudPlatform/testgrid/pb/config"
+	summarypb "github.com/GoogleCloudPlatform/testgrid/pb/summary"
+)
+
+func TestCallOutDifferences(t *testing.T) {
+	passingDashboard := summarypb.DashboardSummary{
+		TabSummaries: []*summarypb.DashboardTabSummary{
+			{
+				DashboardName:    "dash",
+				DashboardTabName: "tab",
+				OverallStatus:    summarypb.DashboardTabSummary_PASS,
+			},
+		},
+	}
+
+	failingDashboard := summarypb.DashboardSummary{
+		TabSummaries: []*summarypb.DashboardTabSummary{
+			{
+				DashboardName:    "dash",
+				DashboardTabName: "tab",
+				OverallStatus:    summarypb.DashboardTabSummary_FAIL,
+				FailingTestSummaries: []*summarypb.FailingTestSummary{
+					{
+						TestName:  "badtest",
+						FailCount: 1,
+					},
+				},
+			},
+		},
+	}
+
+	staleDashboard := summarypb.DashboardSummary{
+		TabSummaries: []*summarypb.DashboardTabSummary{
+			{
+				DashboardName:    "dash",
+				DashboardTabName: "tab",
+				OverallStatus:    summarypb.DashboardTabSummary_STALE,
+			},
+		},
+	}
+
+	testConfig := configpb.Dashboard{
+		DashboardTab: []*configpb.DashboardTab{
+			{
+				Name: "tab",
+				AlertOptions: &configpb.DashboardTabAlertOptions{
+					NumFailuresToAlert: 1,
+				},
+			},
+		},
+		Name: "dash",
+	}
+
+	tests := []struct {
+		name                      string
+		newSummary                *summarypb.DashboardSummary
+		oldSummary                *summarypb.DashboardSummary
+		config                    *configpb.Dashboard
+		expectOnEachFailure       int
+		expectOnEachSuccess       int
+		expectOnConsistentFailure int
+		expectOnBecomingStale     int
+	}{
+		{
+			name:       "No DashboardTabs; No Alert",
+			oldSummary: &summarypb.DashboardSummary{},
+			newSummary: &summarypb.DashboardSummary{},
+			config:     &configpb.Dashboard{},
+		},
+		{
+			name:                "Pass to Pass Dashboards",
```

Comment:
Please state what you are trying to validate -- perhaps "still passing dashboards do not alert?" Just naming the setup isn't as helpful.